bsc.typechecker/higherrank/DeepSeqCond.bs

```diff
@@ -0,0 +1,50 @@
+package DeepSeqCond where
+
+import FoldFIFO
+
+interface Foo =
+  x :: (UInt 8)
+  y :: (Literal a) => a
+
+-- Implicit condition should be surfaced
```

Comment:
So the thing is that this is primDeepSeqCond, not primSeqCond. If you don't want to pick up conditions from the entire structure you should select before trying to surface implicit conditions. But sometimes you want to treat a structure as an opaque object when surfacing conditions (sort of as if you had packed it, but not limited to things in  Bits).

From that perspective, the contract of impCondOf is "give me all of the implicit conditions lurking in this thing" (at least up to function boundaries because introspecting past them is hard and that can't give you accurate post-application conditions in any event). Does that line make sense to you?

Reply:
OK.  I was only thinking about structs as elaboration-time values that get operated on and disappear; I was forgetting the fact that a struct can sometimes just be a bitifiable hardware value.  It only makes sense to call `impCondOf` on expressions that should fully elaborate -- which is why it's OK to give up when we reach a lambda or unconcretized polymorphism -- and structs can be that kind of fully elaborated value (which I was forgetting).

As an aside: I do wonder if there ought to be a warning issued, when `impCondOf` encounters a lambda etc and gives up -- since that would indicate, in my opinion as I said, that `impCondOf` was called in a way that doesn't make sense.

Reply:
I think that I was also remembering that `impCondOf` when applied to a whole interface has given strange results.  But maybe I understand it now: any methods with arguments will be defined as lambdas and not contribute any conditions; any methods of no arguments (and not polymorphic in type) may contribute conditions.  (But I would still avoid `impCondOf` on an interface, as an example of a use that doesn't make sense.)